mxnet-bot/EmailBot/EmailBot.py

```diff
@@ -0,0 +1,324 @@
+        """
+        This method is to count how many pages of issues/labels in total
+        obj could be "issues"/"labels"
+        state could be "open"/"closed"/"all", available to issues
+        """
+        assert obj in set(["issues", "labels"]), "Invalid Input!"
+        url = 'https://api.github.com/repos/{}/{}'.format(self.repo, obj)
+        if obj == 'issues':
+            response = requests.get(url, {'state': state},
+                                    auth=self.auth)
+        else:
+            response = requests.get(url, auth=self.auth)
+        assert response.status_code == 200, response.status_code
+        if "link" not in response.headers:
+            # That means only 1 page exits
+            return 1
+        # response.headers['link'] will looks like:
+        # <https://api.github.com/repositories/34864402/issues?state=all&page=387>; rel="last"
+        # In this case we need to extrac '387' as the count of pages
+        return int(self.__clean_string(response.headers['link'], " ").split()[-3])
+
+    def read_repo(self, periodically=True):
+        """
+        This method is to read issues in the repo.
+        if periodically == True, it will read issues which are created in a specific time period
+        if periodically == False, it will read all issues
+        """
+        logging.info("Start reading {} issues".format("periodically" if periodically else "all"))
+        if periodically:
+            self.__set_period(8)
+        else:
+            self.start = self.sla_start
+            self.end = datetime.datetime.today()+datetime.timedelta(days=2)
+        pages = self.__count_pages('issues', 'all')
+        opendata = []
+        closeddata = []
+        stop = False
+        for page in range(1, pages + 1):
+            url = 'https://api.github.com/repos/' + self.repo + '/issues?page=' + str(page) \
+                  + '&per_page=30'.format(repo=self.repo)
+            response = requests.get(url,
+                                    {'state': 'all',
+                                     'base': 'master',
+                                     'sort': 'created',
+                                     'direction': 'desc'},
+                                    auth=self.auth)
+            response.raise_for_status()
+            for item in response.json():
+                if "pull_request" in item:
+                    continue
+                created = datetime.datetime.strptime(item['created_at'], "%Y-%m-%dT%H:%M:%SZ")
+                if self.start <= created <= self.end:
+                    if item['state'] == 'open':
+                        opendata.append(item)
+                    elif item['state'] == 'closed':
+                        closeddata.append(item)
+                else:
+                    stop = True
+                    break
+            if stop:
+                break
+        self.opendata = opendata
+        self.closeddata = closeddata
+
+    def sort(self):
```

Comment:
Set it private?

Reply:
I think public is better. So that we can not only use this bot to send emails but also to sort issues. For example, if I want to know unlabeled issues in a week. I can simply run:
from EmailBot import EmailBot
eb = EmailBot()
eb.read_repo(True)
data = eb.sort()
print(data['unlabelled'])

Reply:
Can we find a better variable name than "data"? (issues?)